src/renderer/pages/Widget/index.tsx

```diff
@@ -3,15 +3,30 @@ import Header from './Header';
 import SyncAction from './SyncAction';
 import SyncErrorBanner from './SyncErrorBanner';
 import SyncInfo from './SyncInfo';
+import useSyncStatus from '../../hooks/SyncStatus';
+import { useState } from 'react';
+import { SyncStatus } from '../../../main/background-processes/sync';
+import { SyncFailed } from './SyncFailed';
 
 export default function Widget() {
+  const [syncStatus, setSyncStatus] = useState<SyncStatus>('RUNNING');
+  useSyncStatus(setSyncStatus);
+  const handleRetrySync = () => {
```

Comment:
This could be declared on `SyncFailed`, there is no need to pass it as a prop from here

Reply:
note that props are distributed to two childs, SyncFailed and SyncAction component, is just easier to maintain the logic in the parent and distribute it to the childs than distributing logic between childs just to avoid props drilling. 

Now if I need to modify or improve this logic I have everything centralized in this component, the other ones are just dummy components. Do you think it has any benefit to distribute the logic through multiple components in a simple component tree like this one?

Reply:
but handleRetrySync is only passed to SyncFailed or I'm missing something?